Tests/MapboxMapsTests/Foundation/MapViewTests.swift

```diff
@@ -182,9 +188,113 @@ final class MapViewTests: XCTestCase {
         XCTAssertEqual(participant2.participateStub.invocations.count, 1)
     }
 
-    func testMemoryWarning() {
-        // Verifies that a memory warning call is not causing side effects
-        mapView.didReceiveMemoryWarning()
+    func testAppLifecycleNotificationSubscribedWhenDidMoveToNewWindow() {
+        let notificationCenter = MockNotificationCenter()
+        let bundle = MockBundle()
+        bundle.infoDictionaryStub.defaultReturnValue = [:]
+        dependencyProvider.makeNotificationCenterStub.defaultReturnValue = notificationCenter
+        dependencyProvider.makeBundleStub.defaultReturnValue = bundle
+        let mapView = MapView(
+            frame: CGRect(origin: .zero, size: CGSize(width: 100, height: 100)),
+            mapInitOptions: MapInitOptions(),
+            dependencyProvider: dependencyProvider)
+
+        window.addSubview(mapView)
+
+        XCTAssertTrue(notificationCenter.addObserverStub.invocations.contains(where: { stub in
+            stub.parameters.name == UIApplication.didEnterBackgroundNotification
+        }))
+        XCTAssertTrue(notificationCenter.addObserverStub.invocations.contains(where: { stub in
+            stub.parameters.name == UIApplication.willEnterForegroundNotification
+        }))
+        XCTAssertTrue(notificationCenter.addObserverStub.invocations.contains(where: { stub in
+            stub.parameters.name == UIApplication.didReceiveMemoryWarningNotification
+        }))
+        XCTAssertEqual(notificationCenter.addObserverStub.invocations.count, 3)
+    }
+
+    @available(iOS 13.0, *)
+    func testSceneLifecycleNotificationSubscribedWhenDidMoveToNewWindow() {
+        let notificationCenter = MockNotificationCenter()
+        let bundle = MockBundle()
+        bundle.infoDictionaryStub.defaultReturnValue = ["UIApplicationSceneManifest": []]
+        dependencyProvider.makeNotificationCenterStub.defaultReturnValue = notificationCenter
+        dependencyProvider.makeBundleStub.defaultReturnValue = bundle
+        let mapView = MapView(
+            frame: CGRect(origin: .zero, size: CGSize(width: 100, height: 100)),
+            mapInitOptions: MapInitOptions(),
+            dependencyProvider: dependencyProvider)
+
+        window.addSubview(mapView)
+
+        XCTAssertTrue(notificationCenter.addObserverStub.invocations.contains(where: { stub in
+            stub.parameters.name == UIScene.didEnterBackgroundNotification
+        }))
+        XCTAssertTrue(notificationCenter.addObserverStub.invocations.contains(where: { stub in
+            stub.parameters.name == UIScene.willEnterForegroundNotification
+        }))
```

Comment:
We need to verify the `object` as well as the `name` here since an app may contain multiple scenes.